TypeCobol/Compiler/Symbols/NamespaceSymbol.cs

```diff
@@ -102,14 +97,28 @@ public override Scope<ProgramSymbol> Programs
             protected set;
         }
 
-        public override Domain<TypedefSymbol>.Entry ResolveAccessibleType(RootSymbolTable root, string[] path)
+        private Domain<TSymbol>.Entry ResolveSymbol<TSymbol>(string[] path, Func<string, Domain<TSymbol>.Entry> lookupSymbol)
+            where TSymbol : Symbol
         {
-            throw new NotImplementedException();
+            if (path == null || path.Length == 0 || path[0] == null)
+                return null;
+
+            var name = path[0];
+            var results = new Domain<TSymbol>.Entry(name);
+            foreach (var candidate in lookupSymbol(name))
+            {
+                if (candidate.IsMatchingPath(path))
+                {
+                    results.Add(candidate);
+                }
+            }
+
+            return results;
         }
 
         public override Domain<TypedefSymbol>.Entry ResolveType(RootSymbolTable root, string[] path)
         {
-            throw new NotImplementedException();
+            return ResolveSymbol<TypedefSymbol>(path, root.LookupType);
```

Comment:
The idea was to implement the search here and have RootSymbolTable inherit the method but you are right a namespace does not hold any type (despite RootSymbolTable being a special namespace).
I'll rework this class but i'll need to discuss with @mayanje to fully understand what a namespace should be.

Reply:
This has been changed in e7ed72a9ebb5bf0fb057eb3b6e82678da642f3fc but the two methods `ResolveScope` and `ResolveType` are ambiguous because they accept a `RootSymbolTable` argument so :
- either the resolution uses the RootSymbolTable and the process should not depend on the current symbol level (i.e. Program, Namespace or Function should behave the same)
- or the method is really abstract and should not depend on external RootSymbolTable object but rely only on specialized implementations
This should be adressed in PR 3 with the factorization of Resolve* methods I think.